### ISSUE TITLE
GHelper 0.72 failing to autoupdate (crash)

### ISSUE THREAD

**Issue by bel57** [seerge/g-helper#513]
**Describe the bug**

Launching GHelper 0.72 opens a prompt to download the update (0.74). When accepting, it does nothing. Moving the cursor to systray icon and it disappear. The process is not running. Nothing special in GHelper logs. System logs points to a KERNELBASE.dll failure.
This is the first time it's happening to me. I've tried to repeat the operation but it keeps happening, so I'll manually update.
Might be a problem in your code, might be a local issue only with my system. Can you look into this?

**Expected behavior**
Successfully completing the autoupdate.

**App Logs**
[log.txt](https://github.com/seerge/g-helper/files/11632847/log.txt)
[CrashDumps.zip](https://github.com/seerge/g-helper/files/11632851/CrashDumps.zip)

**Screenshots or screencasts**
![image](https://github.com/seerge/g-helper/assets/10614811/3d580ef1-886b-43a6-b014-4eb53a2d4710)

Source
GHelper

Résumé
Fonctionnement arrêté

Date
‎vendredi ‎02/‎06/‎2023 02:58

Statut
Rapport envoyé

Description
Chemin d’accès de l’application défaillante :	C:\Users\Bel\ASUS\GHelper.exe

Signature du problème
Nom d’événement du problème :	APPCRASH
Nom de l’application:	GHelper.exe
Version de l’application:	0.72.0.0
Horodatage de l’application:	641de100
Nom du module défaillant:	KERNELBASE.dll
Version du module défaillant:	10.0.19041.3031
Horodateur du module défaillant:	69314b12
Code de l’exception:	e0434352
Décalage de l’exception:	000000000002cf19
Version du système:	10.0.19045.2.0.0.768.101
Identificateur de paramètres régionaux:	1036
Information supplémentaire n° 1:	00c3
Information supplémentaire n° 2:	00c31b3a8bdc043497ce4fb9a1c58281
Information supplémentaire n° 3:	367d
Information supplémentaire n° 4:	367dbb56656ec84274b4af9893f22d3f

Informations complémentaires sur le problème
ID de compartiment :	8b1394de116eda62881954ffabc2d509 (1736512588403102985)

**Desktop (please complete the following information):**
- OS: Windows 10 22H2
- Laptop model: ROG Zephyrus G14 GA401IU

**Asus software**
- MyASUS installed
- ASUS Optimization service

**Comment by seerge** [comment]
@bel57 usually this crash events come in pair. 1 mentions kernelbase (and is useless, as it doesn't show what went wrong), second one is more specific.

Can you check that ?

**Comment by bel57** [comment]
Sorry the picture/logs are intacts, I'm not hiding any detail about another system event. There is no other crash reported during this period. Only G-Helper when triggering the self update process. Anyway I have manually updated from 0.72 to 0.74. We'll see if it happens again when you release the next stable build.
Thanks for your fast reply.

**Comment by bel57** [comment]
Okay I caught more in the logs, hope that helps

```
Nom du journal :System
Source :       Microsoft-Windows-UserModePowerService
Date :         vendredi 02 06 2023 03:00:27
ID de l’événement :12
Catégorie de la tâche :(10)
Niveau :       Information
Mots clés :    
Utilisateur :  Système
Ordinateur :   BEL-ZEPHYRUS
Description :
Le processus C:\Users\Bel\ASUS\GHelper.exe (ID de processus :5176) a réinitialisé le schéma de stratégie de {de43ed1b-1145-4617-809d-87d59e10ab12} à {de43ed1b-1145-4617-809d-87d59e10ab12}
XML de l’événement :
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Microsoft-Windows-UserModePowerService" Guid="{ce8dee0b-d539-4000-b0f8-77bed049c590}" />
    <EventID>12</EventID>
    <Version>0</Version>
    <Level>4</Level>
    <Task>10</Task>
    <Opcode>0</Opcode>
    <Keywords>0x4000000000000000</Keywords>
    <TimeCreated SystemTime="2023-06-02T07:00:27.8545288Z" />
    <EventRecordID>91072</EventRecordID>
    <Correlation />
    <Execution ProcessID="1668" ThreadID="1112" />
    <Channel>System</Channel>
    <Computer>BEL-ZEPHYRUS</Computer>
    <Security UserID="S-1-5-18" />
  </System>
  <EventData>
    <Data Name="ProcessPath">C:\Users\Bel\ASUS\GHelper.exe</Data>
    <Data Name="ProcessPid">5176</Data>
    <Data Name="OldSchemeGuid">{de43ed1b-1145-4617-809d-87d59e10ab12}</Data>
    <Data Name="NewSchemeGuid">{de43ed1b-1145-4617-809d-87d59e10ab12}</Data>
  </EventData>
</Event>
```

`Nom du journal :Application
Source :       .NET Runtime
Date :         vendredi 02 06 2023 03:00:30
ID de l’événement :1026
Catégorie de la tâche :Aucun
Niveau :       Erreur
Mots clés :    Classique
Utilisateur :  N/A
Ordinateur :   BEL-ZEPHYRUS
Description :
Application: GHelper.exe
CoreCLR Version: 7.0.523.17405
.NET Version: 7.0.5
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Net.WebException: An exception occurred during a WebClient request.
 ---> System.IO.IOException: The process cannot access the file 'C:\Users\Bel\ASUS\GHelper.exe' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   --- End of inner exception stack trace ---
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at GHelper.SettingsForm.AutoUpdate(String requestUri) in C:\Users\Serge\g-helper\app\Settings.cs:line 417
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()

XML de l’événement :
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name=".NET Runtime" />
    <EventID Qualifiers="0">1026</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2023-06-02T07:00:30.4511951Z" />
    <EventRecordID>125037</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>BEL-ZEPHYRUS</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Application: GHelper.exe
CoreCLR Version: 7.0.523.17405
.NET Version: 7.0.5
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Net.WebException: An exception occurred during a WebClient request.
 ---&gt; System.IO.IOException: The process cannot access the file 'C:\Users\Bel\ASUS\GHelper.exe' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   --- End of inner exception stack trace ---
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at GHelper.SettingsForm.AutoUpdate(String requestUri) in C:\Users\Serge\g-helper\app\Settings.cs:line 417
   at System.Threading.Tasks.Task.&lt;&gt;c.&lt;ThrowAsync&gt;b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
</Data>
  </EventData>
</Event>`

**Comment by seerge** [comment]
@bel57 reason was in github release configuration, i have fixed it here and it should work now (please try) :)

**Comment by bel57** [comment]
Thanks for the quick fix, it worked just fine :)

**Comment by seerge** [comment]
@bel57 thanks for spotting btw. Reason was that it tried to download .exe variant from github (instead of zip) and unzip it (good luck with that).

I have added a "permament" fix in 0.76 for this. But for now, releases will be in zips )))
